core-bundle/src/Routing/UrlGenerator.php

```diff
@@ -173,7 +177,7 @@ private function addHostToContext(RequestContext $context, array $parameters, in
         $context->setHost($host);
         $referenceType = UrlGeneratorInterface::ABSOLUTE_URL;
 
-        if (!$port) {
+        if (null === $port) {
```

Comment:
Why this?

Reply:
Because now `$port` is a nullable `int`. Before it was a `string` / `empty string`.

Reply:
Port `0` would behave differently now, is that on purpose?

Reply:
Good question. IIRC you can bind to port 0 e.g. when using sockets (which means: connect to any free port) but it's not a valid TCP/UDP port per se. I'm not sure this edge case matters. But I can revert this change oc if you prefer.

Reply:
I am for reverting the change.

Reply:
I am for reverting the change.

Reply:
Reverted in 77e2a4dc9e31bdb14f4a736ace119e5223052bc7.